splunk-otel-android/src/test/java/com/splunk/rum/RumResponseAttributesExtractorTest.java

```diff
@@ -0,0 +1,100 @@
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class RumResponseAttributesExtractorTest {
+    @Rule
+    public OpenTelemetryRule otelTesting = OpenTelemetryRule.create();
+    private Tracer tracer;
+
+    @Before
+    public void setup() {
+        tracer = otelTesting.getOpenTelemetry().getTracer("testTracer");
+    }
+
+    @Test
+    public void spanDecoration() {
+        ServerTimingHeaderParser headerParser = mock(ServerTimingHeaderParser.class);
+        when(headerParser.parse("headerValue")).thenReturn(new String[]{"9499195c502eb217c448a68bfe0f967c", "fe16eca542cd5d86"});
+
+        Request fakeRequest = mock(Request.class);
+        Response response = new Response.Builder()
+                .request(fakeRequest)
+                .protocol(Protocol.HTTP_1_1)
+                .message("hello")
+                .code(200)
+                .addHeader("Server-Timing", "headerValue")
+                .addHeader("Content-Length", "101")
+                .build();
+
+        RumResponseAttributesExtractor attributesExtractor = new RumResponseAttributesExtractor(headerParser);
+        AttributesBuilder attributesBuilder = Attributes.builder();
+        attributesExtractor.onStart(attributesBuilder, fakeRequest);
+        attributesExtractor.onEnd(attributesBuilder, fakeRequest, response);
+        Attributes attributes = attributesBuilder.build();
+
+        assertEquals("http", attributes.get(SplunkRum.COMPONENT_KEY));
+        assertEquals("9499195c502eb217c448a68bfe0f967c", attributes.get(OkHttpRumInterceptor.LINK_TRACE_ID_KEY));
+        assertEquals("fe16eca542cd5d86", attributes.get(OkHttpRumInterceptor.LINK_SPAN_ID_KEY));
+        assertEquals(101L, (long) attributes.get(HTTP_RESPONSE_CONTENT_LENGTH));
+    }
+
+    @Test
+    public void spanDecoration_noHeader() throws IOException {
+        ServerTimingHeaderParser headerParser = mock(ServerTimingHeaderParser.class);
+        when(headerParser.parse(null)).thenReturn(new String[0]);
+
+        Request fakeRequest = mock(Request.class);
+        Response response = new Response.Builder()
+                .request(fakeRequest)
+                .protocol(Protocol.HTTP_1_1)
+                .message("hello")
+                .code(200)
+                .build();
+
+        RumResponseAttributesExtractor attributesExtractor = new RumResponseAttributesExtractor(headerParser);
+        AttributesBuilder attributesBuilder = Attributes.builder();
+        attributesExtractor.onEnd(attributesBuilder, fakeRequest, response);
+        attributesExtractor.onStart(attributesBuilder, fakeRequest);
+        Attributes attributes = attributesBuilder.build();
+
+        assertEquals("http", attributes.get(SplunkRum.COMPONENT_KEY));
+        assertNull(attributes.get(OkHttpRumInterceptor.LINK_TRACE_ID_KEY));
+        assertNull(attributes.get(OkHttpRumInterceptor.LINK_SPAN_ID_KEY));
```

Comment:
Nit: Might be worth also verifying that the content length is still recorded even without this special header?

Reply:
added. good call